server/src/controllers/job_posts/getOneJobPost.js

```diff
@@ -0,0 +1,42 @@
+const { query } = require("express");
+const JobPosting = require("../../models/job_posting.model");
+const logger = require("pino")();
+/**
+a) GET request (get a job post’s information, get all columns based on id) using the job post id parameter
+Return all info about that specific job post
+ */
+
+const getJobPostsRequestHandler = async (req, res) => {
+    // check method is GET
+    if (req.method !== 'GET') {
+        return res.status(405).json({ message: 'Method not allowed, only GET methods allowed.' });
+    }
+
+    try{
+        // ------ Get Job Id
+        const jobPostId = req?.query?.id ? parseInt(req.query.id, 10) : null;
```

Comment:
I'm not too sure about this but checking if the correct datatype for the given jobPostId might be good. I think setting it to NULL will make the query search for NULL Job posting IDs, which will return an empty query. If this is the intended functionality, feel free to ignore what I said!